cms/apps/media/models.py

```diff
@@ -212,10 +318,103 @@ class Video(models.Model):
+        if (self.high_resolution_mp4 or self.low_resolution_mp4) and self.external_video:
+            raise ValidationError({
+                'high_resolution_mp4': "Please provide either a locally hosted file or an external file, not both."
+            })
+
+        if self.external_video:
+            info = get_video_info(self.external_video)
+            if info:
+                self.external_video_iframe_url = info['src']
+
+                if not self.external_video_iframe_url:
+
+                    raise ValidationError({
+                        'external_video': "Couldn't determine how to embed this video. Maybe the video's privacy settings disallow embedding?"
+                    })
+                self.external_video_id = info['id']
+                self.external_video_service = info['service']
+
+        return super().clean()
+
+    def embed_html(self, loop=False, autoplay=False, controls=False, mute=False, youtube_parameters=None, width=640, height=360):
+        '''
+        Returns the HTML code for embedding the video.
+        Expects youtube_parameters as a dictionary in the form {parameter:value}
+        When using, this is a function so call with {{ video.embed_html|safe }}
+        '''
+        if self.external_video:
+            if self.external_video_service == 'youtube':
+                return '<iframe class="vid-Video" frameborder="0" allowfullscreen="1" allow="autoplay; ' \
+                       'encrypted-media" title="YouTube video player"{}{} ' \
+                       'src="{}?autoplay={}&amp;controls={}&amp;loop={}&amp;mute={}{}"></iframe>'.format(
+                    ' width="{}"'.format(width) if width else '',
+                    ' height="{}"'.format(height) if height else '',
+                    self.external_video_iframe_url,
+                    int(autoplay),
+                    int(controls),
+                    int(loop),
+                    int(mute),
+                    ('&amp;' + '&amp;'.join('{}={}'.format(parameter,youtube_parameters[parameter]) for parameter in youtube_parameters)) if youtube_parameters else '',
+                )
+            elif self.external_video_service == 'vimeo':
+                return '<iframe class="vid-Video" frameborder="0" allowfullscreen="1" title="Vimeo video player" ' \
+                       'width="{}" height="{}" allowfullscreen ' \
+                       'src="{}?autoplay={}&amp;background={}&amp;loop={}&amp;muted={}"></iframe>'.format(
+                    width,
+                    height,
+                    self.external_video_iframe_url,
+                    int(autoplay),
+                    int(controls),
+                    int(loop),
+                    int(mute),
+                )
+        elif self.high_resolution_mp4 or self.low_resolution_mp4:
+            return '<video preload="{}" {}{}{}{} class="vid-Video">' \
+                   '<source src="{}" type="video/mp4" media="all and (max-width:480px)">' \
```

Comment:
media= is deprecated, and doesn't work at all in Chrome. The likely case is that Chrome will just see the first video in a format it recognises and play that, which means it will always play the lowest resolution one.

Reply:
What's the use of low_res_file then? The only other way is to use js to change the video src on the fly